Docker.md

```diff
@@ -2,8 +2,19 @@
 
 ### Publishing ports
 
-Ports for services should be explicitly published to the loopback interface, 127.0.0.1 to prevent docker from binding them to 0.0.0.0.
-For example, when using postgres or elasticsearch, we might want to make them available to developers, but in the case that we deploy on a public server for testing we don't want them to be exposed.
+By default, containers within the same docker-compose network are able to connect to services running
+on other containers on any port by using the container name, as long as the container is listening
+on 0:0:0:0.
+
+The [`EXPOSE` directive isn't needed](https://docs.docker.com/engine/reference/builder/#expose), although
+some tools ([such as the nginx proxy companion](https://github.com/nginx-proxy/acme-companion#step-3---proxied-containers))
+use it to define a contract between different containers. Use it if you think it will be necessary.
+
+When publishing services in a docker-compose.yaml file, ports should be explicitly published to the
+loopback interface, 127.0.0.1 to prevent docker from automatically binding them to 0.0.0.0
+(all interfaces, open to the world).
```

Comment:
Isn’t using `expose` instead of `ports` preventing to automatically bind them to `0.0.0.0`?

Reply:
good question. without testing I think the answer is no. As the docker documentation says, expose has no effect on the port being available on the _host_. You _must_ use a port declaration for this. By default if you don't specify a host in this declaration it will use 0.0.0.0. I will double-check this now.

Reply:
From https://docs.docker.com/compose/compose-file/compose-file-v3/#expose

> Expose ports without publishing them to the host machine - they’ll only be accessible to linked services. Only the internal port can be specified.

Reply:
this is still confusing to me, because it is not necessary to use `- expose` or `EXPOSE` to be able to access a port from a different service/container. It's possible that this is from old versions of docker compose where `- links` was a required field, but in my experience you can listen on a port and connect to it without exposing it.